packages/content/src/enrichments.ts

```diff
@@ -0,0 +1,443 @@
+  const metadataUpdatedDate = parseISODateString(
+    getValueFromMetadata(newContent, "metadataUpdatedDate")
+  );
+  if (metadataUpdatedDate) {
+    newContent.metadataUpdatedDate = metadataUpdatedDate.date;
+    newContent.metadataUpdatedDatePrecision = metadataUpdatedDate.precision;
+    newContent.metadataUpdatedDateSource = getMetadataPath(
+      "metadataUpdatedDate"
+    );
+  } else {
+    newContent.metadataUpdatedDate = newContent.updatedDate;
+    newContent.metadataUpdatedDatePrecision = DatePrecision.Day;
+    newContent.metadataUpdatedDateSource = newContent.updatedDateSource;
+  }
+
+  // updatedDate & updatedDateSource:
+  // updatedDate is already set to item.modified, we will override that if we have reviseDate in metadata or lastEditDate
+  const reviseDate = parseISODateString(
+    getValueFromMetadata(newContent, "reviseDate")
+  );
+  const layerLastEditDate = getProp(
+    newContent,
+    "layer.editingInfo.lastEditDate"
+  );
+  const serverLastEditDate = getProp(
+    newContent,
+    "server.editingInfo.lastEditDate"
+  );
+  newContent.updatedDatePrecision = DatePrecision.Day;
+  if (reviseDate) {
+    newContent.updatedDate = reviseDate.date;
+    newContent.updatedDatePrecision = reviseDate.precision;
+    newContent.updatedDateSource = getMetadataPath("reviseDate");
+  } else if (layerLastEditDate) {
+    newContent.updatedDate = new Date(layerLastEditDate);
+    newContent.updatedDateSource = "layer.editingInfo.lastEditDate";
+  } else if (serverLastEditDate) {
+    newContent.updatedDate = new Date(serverLastEditDate);
+    newContent.updatedDateSource = "server.editingInfo.lastEditDate";
+  }
+
+  // publishedDate & publishedDateSource:
+  // publishedDate is already set to item.created, we will override that if we have pubDate or createdDate in metadata
+  const pubDate = parseISODateString(
+    getValueFromMetadata(newContent, "pubDate")
+  );
+  const createDate = parseISODateString(
+    getValueFromMetadata(newContent, "createDate")
+  );
+  newContent.publishedDatePrecision = DatePrecision.Day;
+  if (pubDate) {
+    newContent.publishedDate = pubDate.date;
+    newContent.publishedDatePrecision = pubDate.precision;
+    newContent.publishedDateSource = getMetadataPath("pubDate");
+  } else if (createDate) {
+    newContent.publishedDate = createDate.date;
+    newContent.publishedDatePrecision = createDate.precision;
+    newContent.publishedDateSource = getMetadataPath("createDate");
+  }
+
+  return newContent;
+}
+
+const fetchGroupIds = (
+  content: IHubContent,
+  requestOptions?: IHubRequestOptions
+): Promise<string[]> => {
+  return getItemGroups(content.id, requestOptions).then(response => {
+    const { admin, member, other } = response;
+    return [...admin, ...member, ...other].map(group => group.id);
+  });
+};
+
+const fetchMetadata = (
+  content: IHubContent,
+  requestOptions?: IHubRequestOptions
+) => getContentMetadata(content.id, requestOptions);
+
+const fetchOwnerOrgId = (
```

Comment:
can we expose this? I need for [PR-9667](https://github.com/ArcGIS/opendata-ui/pull/9667/files#diff-076d4969300d0bb9d112746b1dfc71f68ff2e0abe1ecde160cd68e21885f4f6fR258) unless changes in this PR will result in `orgId` being present on content created from `composer` datasets.

Reply:
so, I swear I left a comment about this last night, but it seems to have gotten lost.

I definitely can, but you can also achieve the same w/ `const { orgId } = fetchEnrichments(content, { ...requestOptions, enrichments: ['orgId'] }`

That is a little awkward, so if you think it's better just to expose each function, I can do that too.